### PR TITLE
Add VSCode support for CS149

### LINKED ISSUE #315: Need to install VSCode for CS149
VSCode, Java Extension Pack, and Checkstyle for Java need to be installed for Dr. Bowers.

### PR DESCRIPTION
**Pull request by ripleymj**
Adds Microsoft key and repo, installs Code, and adds Java/Checkstyle plugins per user. Like Eclipse, it leaves a checkstyle.xml configuration to the user.

Closes #315

### CODE REVIEW

File: roles/vscode/files/vscode.list [hunk 1]
```diff
@@ -0,0 +1 @@
+deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main
```
Comment (kylelaker) [inline roles/vscode/files/vscode.list]: I know they recommend specifying the architecture, but is it really necessary if amd64 is the only one we support anyway?

File: roles/vscode/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+# tasks file for vscode
+- name: Add Microsoft key
+  apt_key:
+    id: BC528686B50D79E339D3721CEB3E94ADBE1229CF
+    url: https://packages.microsoft.com/keys/microsoft.asc
+    state: present
+- name: Install VSCode Repository file
```
Comment (kylelaker) [inline roles/vscode/tasks/main.yml]: Is there a particular preference for this implementation over the following?

```
- apt_repository:
    repo: deb https://packages.microsoft.com/repos/vscode stable main
    state: present
    filename: vscode
```

Comment (kylelaker) [reply roles/vscode/tasks/main.yml]: FWIW, an `apt update` is automatically performed when an `apt_repository` entry changes by default.

File: roles/vscode/files/vscode.list (hunk 1 above)
Comment (ripleymj) [reply roles/vscode/files/vscode.list]: Only because Microsoft provides it that way.

File: roles/vscode/tasks/main.yml (hunk 2 above)
Comment (ripleymj) [reply roles/vscode/tasks/main.yml]: Stole this from robots, but there we're using a template to set the Ubuntu version, so I guess it's less necessary here. And I didn't realize apt-repository was a thing.

Comment (kylelaker) [reply roles/vscode/tasks/main.yml]: Yeah, I am a little curious if we should evaluate our use of .list files. Think in most the cases we could use the variable expansion except maybe common where we explicitly want to override entire files... But resolving this in other places can be a separate issue at a later time.

File: roles/vscode/files/vscode.list (hunk 1 above)
Comment (kylelaker) [inline roles/vscode/files/vscode.list]: I think this file can be removed now

Comment (ripleymj) [reply roles/vscode/files/vscode.list]: Derp

**Pull request merged**